crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -323,8 +322,13 @@ where
     /// If automatically inserting [`apply_deferred`](crate::schedule::apply_deferred) like
     /// this isn't desired, use [`after_ignore_deferred`](Self::after_ignore_deferred) instead.
     ///
-    /// Note: The given set is not implicitly added to the schedule when this system set is added.
-    /// It is safe, but no dependencies will be created.
+    /// # Note
+    ///
+    /// If you configure two groups of systems (e.g. 'Set A') in a schedule, and another group references `.after` or `.before` (e.g. 'Set B'), the systems in Set B will not be automatically scheduled.
+    ///
+    /// This means that Set A and Set B will run independently unless they are explicitly linked. So `.after`/`.before` will not provide the desired behaviour, so the Sets can run in parallel or in any order determined by the scheduler.
+    ///
+    /// If you're adding systems in the same location, it's recommended that you use [**`.chain`**](Self::chain) for brevity and clarity. You can also add a set of systems. You may want to use `.after` or `.before` in association with a comment when the reasoning needs to be particularly clear and the order of execution is important or not immediately obvious.
```

Comment:
```suggestion
```

Reply:
This is a bit of a different topic, but since it's not mentioned anywhere else, I think it makes sense to add it to the caveats here. The old text has been removed because I don't think it is needed anymore, given the other changes we made.

Reply:
Can we **remove the dashes**? 

something like:

```
/// any ordering calls between them, whether using `.before`, `.after,` or `.chain`, will be silently ignored.
```

Reply:
If you like. They're called [em dashes](https://www.grammarly.com/blog/why-you-should-love-the-em-dash/) and not uncommon in writing, but I'll leave the choice up to you. Changing my suggestion to that you can apply it.

Reply:
Update: changed the suggestion :)

Reply:
Thanks for explaining. In that case, let's keep it no? There was a type on my side ...  the comma in `.after,`. I'll fix it and let's keep dashes ;)
And hopefully it's finally done.
I learned a lot about scheduling and it was worth. This piece is very important for bevy users.

Reply:
Hehe, sure. Updated the suggestion again :)
Yeah, I think the back and forth really illustrates how incredibly subtle this footgun is!